dbms/src/Common/MyTime.cpp

```diff
@@ -553,6 +553,15 @@ const String & MyTimeBase::monthName() const
 }
 
 bool checkTimeValid(Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second)
+{
+    if (year > 9999 || month <= 0 || month > 12 || day <= 0 || day > 31 || hour > 23 || minute > 59 || second > 59)
+    {
+        return false;
+    }
+    return day <= getLastDay(year, month);
+}
+
+bool checkTimeValidAllowMonthAndDayZero(Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second)
```

Comment:
Better to reuse checkTimeValid, to avoid so many detailed check twice.

Reply:
done.